RELEASE.md

```diff
@@ -1,30 +1,55 @@
-    i. Cut a new release branch from the existing release branch
-    ii. Make your changes
-    iii. Bump the `-rc` version (e.g. `-rc0` -> `-rc1`)
-    iv. Continue to Step 3
-3. Run CI tests on release branch
-4. Update [CHANGELOG.md](https://github.com/Datatamer/unify-client-python/blob/master/CHANGELOG.md) with new entry containing:
-    - Release version
-    - Date of release
-    - `ADDED` (only if applicable)
-    - `REMOVED` (only if applicable)
-    - `FIXED` (only if applicable)
-5. Build release candidate artifacts
-  `python setup.py sdist bdist_wheel`
-6. Test publish to test.pypi.org
-  `twine upload --repository-url https://test.pypi.org/legacy/ dist/*<rc version>*`
-7. Remove `-rc` suffix from version in [VERSION.txt](https://github.com/Datatamer/unify-client-python/blob/master/VERSION.txt)
-8. Merge release branch onto `master`
-9. Tag latest commit with version number on Github
-  `git tag -a v<version> -m "Release v<version>"`
-  `git push Datatamer --tags`
-10. Build release artifacts
-  `python setup.py sdist bdist_wheel`
-11. Publish to PyPI
-  `twine upload dist/*<version>*`
-    - If necessary, update the version/date of the corresponding entry in [CHANGELOG.md](https://github.com/Datatamer/unify-client-python/blob/master/CHANGELOG.md)
+During the following steps we will releasing the `0.3.0` version as an example.
+
+For our example, that means `master` branch is currently on version `0.3.0-dev` (you can check `VERSION.txt` to see the version).
+
+Be sure to substitute `0.3.0` appropriately for the actual version being released.
+
+# 1. Version bump on `master`
+
+Create a PR with the following changes:
+- `Version.txt`: bump the version to the next one, keeping the `-dev` suffix e.g. `0.3.0-dev` -> `0.4.0-dev`.
+- `CHANGELOG.md`:
+  - Create a new section for the new development version e.g. Add `# 0.4.0-dev` to the top of the changelog (with an empty line between it and the next version header).
+
+Merge your changes into `master`.
+
+# 2. Cut a release branch
+
+On the [Datatamer/unify-client-python](https://github.com/Datatamer/unify-client-python) Github repo, click on [Commits](https://github.com/Datatamer/unify-client-python/commits/master). Navigate to the commit just before the version bump commit from Step 1. Click the `<>` icon to browse the repo at that commit.
+
+Then, create a branch on Github within the [Datatamer/unify-client-python](https://github.com/Datatamer/unify-client-python) repo titled `release-<version>` e.g. `release-0.3.0`.
+
+NOTE: This release branch should *not* contain the version bump changes from Step 1.
+
+# 3. Remove `-dev` suffix on release branch
+
+Create a PR *to the release branch* with the following changes:
+- `VERSION.txt`: Remove `-dev` suffix from version e.g. `0.3.0-dev` -> `0.3.0`.
+- `CHANGELOG.md`: Remove the `-dev` suffix from the version being released e.g. `# 0.3.0-dev` -> `# 0.3.0`.
+
+Merge your changes into the release branch e.g. `release-0.3.0`.
+
```

Comment:
don't we want to mention checking the Travis test results before creating the release?